tests/gradients/test_parameter_shift.py

```diff
@@ -28,52 +28,53 @@ def test_non_differentiable(self):
         correctly marked"""
         psi = np.array([1, 0, 1, 0]) / np.sqrt(2)
 
-        with qml.tape.JacobianTape() as tape:
+        with qml.tape.QuantumTape() as tape:
             qml.QubitStateVector(psi, wires=[0, 1])
             qml.RX(0.543, wires=[0])
             qml.RY(-0.654, wires=[1])
             qml.CNOT(wires=[0, 1])
             qml.probs(wires=[0, 1])
 
-        _gradient_analysis(tape)
+        gradient_analysis(tape)
 
         assert tape._par_info[0]["grad_method"] is None
         assert tape._par_info[1]["grad_method"] == "A"
         assert tape._par_info[2]["grad_method"] == "A"
 
     def test_analysis_caching(self, mocker):
-        """Test that the gradient analysis is only executed once per tape"""
+        """Test that the gradient analysis is only executed once per tape
+        if grad_fn is set an unchanged."""
         psi = np.array([1, 0, 1, 0]) / np.sqrt(2)
 
-        with qml.tape.JacobianTape() as tape:
+        with qml.tape.QuantumTape() as tape:
             qml.QubitStateVector(psi, wires=[0, 1])
             qml.RX(0.543, wires=[0])
             qml.RY(-0.654, wires=[1])
             qml.CNOT(wires=[0, 1])
             qml.probs(wires=[0, 1])
 
         spy = mocker.spy(qml.operation, "has_grad_method")
-        _gradient_analysis(tape)
+        gradient_analysis(tape, grad_fn=5)

```

Comment:
Why pass `grad_fn=5`?

Reply:
`gradient_analysis` has behaviour that caches the analysis iff:

- the tape is the same
- the `grad_fn` is the same.

This way, you can run `gradient_analysis(tape, param_shift)` multiple times while avoiding overhead. But if you _change_ the gradient function, then the analysis will be re-run.

It is a bit ugly, and the design of `gradient_analysis` should be re-thought, but I thought out of scope for this PR.

> Why pass grad_fn=5?

I thought it would be better test design then doing `grad_fn=qml.gradients.param_shift`, which could be an annoying point of test failure if we move/change this function in the future!